packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -0,0 +1,146 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as fs from 'fs'
+import { getLogger, Logger } from '../../shared/logger'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { VueWebview } from '../../webviews/main'
+import { ExtContext } from '../../shared/extensions'
+//import { telemetry } from '../../shared/telemetry/telemetry'
+import { AccessAnalyzer, S3 } from 'aws-sdk'
+import { exec } from 'child_process'
+const AmazonS3URI = require('amazon-s3-uri')
+
+export interface IamPolicyChecksInitialData {
+    referenceFilePath: string
+    cfnParameterPath: string
+    referenceDocument: string
```

Comment:
If I read it correct, referenceFilePath is from customer input whereas referenceDocument is from reading the file. Can we add some comments to explain this?

Reply:
should we mark this as readonly?

Reply:
Yes, I think we can mark as readonly and I will add comments to denote that.